lib/ClangImporter/ClangImporter.cpp

```diff
@@ -4053,28 +4067,25 @@ ClangDirectLookupRequest::evaluate(Evaluator &evaluator,
       getClangOwningModule(clangDecl, clangDecl->getASTContext());
   auto *lookupTable = ctx.getClangModuleLoader()->findLookupTable(clangModule);
 
-  auto *swiftDeclContext = desc.decl->getInnermostDeclContext();
-  auto *declContextTypeDecl = swiftDeclContext->getSelfNominalTypeDecl();
-  auto effectiveClangContext =
-      ctx.getClangModuleLoader()->getEffectiveClangContext(declContextTypeDecl);
-
   auto foundDecls = lookupTable->lookup(
-      SerializedSwiftName(desc.name.getBaseName()), effectiveClangContext);
+      SerializedSwiftName(desc.name.getBaseName()), EffectiveClangContext());
   // Make sure that `clangDecl` is the parent of all the members we found.
   SmallVector<SwiftLookupTable::SingleEntry, 4> filteredDecls;
-  llvm::copy_if(
-      foundDecls, std::back_inserter(filteredDecls),
-      [clangDecl](SwiftLookupTable::SingleEntry decl) {
-            auto first = decl.get<clang::NamedDecl *>()->getDeclContext();
-            auto second = cast<clang::DeclContext>(clangDecl);
-            if (auto firstDecl = dyn_cast<clang::Decl>(first)) {
-              if (auto secondDecl = dyn_cast<clang::Decl>(second))
-                return firstDecl->getCanonicalDecl() == secondDecl->getCanonicalDecl();
-              else
-                return false;
-            }
-            return first == second;
-      });
+  llvm::copy_if(foundDecls, std::back_inserter(filteredDecls),
+                [clangDecl](SwiftLookupTable::SingleEntry decl) {
+                  auto foundClangDecl = decl.dyn_cast<clang::NamedDecl *>();
```

Comment:
Did you run into problems with this? Macros can't be members so this should be fine.

Reply:
yes, this broke the stdlib build.

Reply:
Hmm, that's super interesting. Do you know how it broke the build or what decl it was? I might go investigate after this lands..

Reply:
it was `p_starttime`

Reply:
which I believe is a macro

Reply:
That's super interesting... so it's a macro defined to a member of a struct. I guess this is the correct way to handle this...

Reply:
I wish this could just be a `ranges::filter(isa-named-decl) | ranges::filter(...)`